src/QMCWaveFunctions/detail/CUDA/cuBLAS_LU.cu

```diff
@@ -66,35 +65,38 @@ __global__ void computeLogDet_kernel(const int n,
                                      cuDoubleComplex* logdets)
 {
   const int iw         = blockIdx.x;
-  const int block_num  = blockIdx.y;
   const T* lu_iw       = mat_lus[iw];
   const int* pivots_iw = pivots + iw * n;
+
   __shared__ cuDoubleComplex logdet_vals[COLBS];
-  logdet_vals[threadIdx.x].x = 0.0;
-  logdet_vals[threadIdx.x].y = 0.0;
-  T lu_diag;
-  int n_index = threadIdx.x + block_num * COLBS;
-  if (n_index < n)
+  const int tid      = threadIdx.x;
+  logdet_vals[tid].x = 0.0;
+  logdet_vals[tid].y = 0.0;
+
+  const int num_col_blocks = (n + COLBS - 1) / COLBS;
```

Comment:
This needs something like
```c++
static_assert(COLBS % 2 == 0);
```
And a warning for the unwary that this reduction stratedgy only works for powers of 2.  Also do you have profiling information of how much this is worth?  While I too believe it should be faster that's just speculation.

Reply:
I added a static assert to check exactly power of 2. I have not not looked at the performance but it should be anyway cheaper than getrf.